src/workerd/io/io-context.c++

```diff
@@ -552,6 +558,23 @@ class IoContext::PendingEvent: public kj::Refcounted {
 };
 
 IoContext::~IoContext() noexcept(false) {
+  // Dispatch unload events if any were attached
+  if (!unloadListeners.empty()) {
+    runInContextScope(
+        Worker::Lock::TakeSynchronously(kj::none), kj::none, [&](Worker::Lock& workerLock) {
+      // Suppress IoContext to prevent unload handlers from scheduling new IO
+      SuppressIoContextScope suppressIo;
+      auto unloadEvent = jsg::alloc<api::Event>("unload");
+      for (auto& listener: unloadListeners) {
+        try {
+          listener(workerLock, unloadEvent.addRef());
```

Comment:
If one of these throws a fatal exception, or we hit something like a OOM, then we really shouldn't try to continue. I think you might need to wrap this in a `js.tryCatch(...)`, and I'd consider at least logging the error rather than ignoring it entirely.

Reply:
The error is propagated as an unhandled exception which is then logged and this is tested. I suspect using tryCatch would inhibit that?